packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -953,7 +953,14 @@ class SelectionOverlay {
        _lineHeightAtEnd = lineHeightAtEnd,
        _selectionEndpoints = selectionEndpoints,
        _toolbarLocation = toolbarLocation,
-       assert(debugCheckHasOverlay(context));
+        hapticFeedbackEnabled = true,
+        assert(debugCheckHasOverlay(context)) {
+    unawaited(() async {
+      // Overrite the default value with the actual value when it is available.
+      final bool? newValue = await HapticFeedback.isHapticFeedbackEnabled();
```

Comment:
Should this be something stored in binding? Since this is a boolean and we don't need to fetch every time.

Reply:
You can add a value in https://api.flutter.dev/flutter/semantics/SemanticsBinding/accessibilityFeatures.html

It already does some caching in there. feel free to schedule a meeting if you would like to pair

Reply:
Notes for me to do a rewrite. Instead of putting this ability into the engine haptics class it should go into accessibility. Here are some interesting classes to examine. 

AccessibilityFeatures in AccessabilityBridge.java 
engine/lib/ui/Window.dart 
bindings.dart has class accessabilityFeatures
SemanticsBinding.instance.accessabiliyFeatures.<new thing>